control-plane-agent/internal/agent.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+package internal
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	controlplane "control-plane-agent/api/control-plane"
+	"control-plane-agent/api/proxy"
+	"control-plane-agent/internal/event"
+	"control-plane-agent/internal/logic"
+	"control-plane-agent/internal/registry"
+)
+
+type Config struct {
+	IsProd bool
+}
+
+func RunAgent() {
+	// DEBUG
+	cfg := Config{IsProd: false}
+
+	// Setup logger
+	if cfg.IsProd {
+		logrus.SetLevel(logrus.InfoLevel)
+		logrus.SetFormatter(&logrus.JSONFormatter{TimestampFormat: time.RFC3339Nano})
+	} else {
+		logrus.SetLevel(logrus.TraceLevel)
+		logrus.SetFormatter(&logrus.TextFormatter{ForceColors: true})
+	}
+
+	// Setup logger for http standard output
+	log.SetOutput(logrus.StandardLogger().Writer())
+
+	logrus.Info("Mesh Control Plane Agent started")
+
+	controlPlaneAPI := controlplane.NewAPI(controlplane.Config{ListenPort: 8100})
```

Comment:
Don't hardcode ports, optionally move to 'descriptive' variables

Reply:
You're totally right. But I have to leave it as it is for now because the work is still in progress. The hardcoded values like this one will be eliminated in the following PRs.